File: recipe-management-domain/src/main/java/fr/cta/recipe/management/domain/entity/IngredientUnit.java
package fr.cta.recipe.management.domain.entity;

import lombok.EqualsAndHashCode;
import lombok.Value;

import javax.validation.constraints.NotBlank;
import java.util.UUID;

@Value
@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
public class IngredientUnit extends Entity<UUID> {

    @NotBlank
    String name;

    public IngredientUnit(UUID id, String name) {
        super(id);
        this.name = name;

        validate();
    }

    public IngredientUnit(String name) {
        this(UUID.randomUUID(), name);
    }
}

File: recipe-management-domain/src/main/kotlin/fr/cta/recipe/management/domain/entity/IngredientUnit.kt
package fr.cta.recipe.management.domain.entity

import com.benasher44.uuid.Uuid
import fr.cta.recipe.management.domain.validation.Validation

data class IngredientUnit(
        override val id: Uuid,
        val name: String) : Entity(id) {

    init {
        Validation.validateNotBlank(::name)
    }

    constructor(name: String): this(Uuid.randomUUID(), name)
}